tests/test_srv6.py

```diff
@@ -627,6 +629,369 @@ def test_srv6(self, dvs, testlog):
+
+        self.setup_srv6(dvs)
+
+        # configure srv6 locator
+        dvs.runcmd("vtysh -c \"configure terminal\" -c \"segment-routing\" -c \"srv6\" -c \"locators\" -c \"locator loc1\" -c \"prefix fc00:0:0:1::/64 block-len 40 node-len 24 func-bits 16\"")
+
+        # create srv6 mysid end.dt6 behavior
+        dvs.runcmd("ip -6 route add fc00:0:0:1:6b::/128 encap seg6local action End.DT6 vrftable {} dev sr0".format(self.vrf_table_id))
+
+        # check application database
+        self.pdb.wait_for_entry("SRV6_MY_SID_TABLE", "40:24:16:0:fc00:0:0:1:6b::")
+
+        # verify that the mysid has been programmed into the ASIC
+        self.adb.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_MY_SID_ENTRY", len(self.initial_my_sid_entries) + 1)
+
+        # check ASIC SAI_OBJECT_TYPE_MY_SID_ENTRY database
+        my_sid = get_created_entry(self.adb.db_connection, "ASIC_STATE:SAI_OBJECT_TYPE_MY_SID_ENTRY", self.initial_my_sid_entries)
+        tbl = swsscommon.Table(self.adb.db_connection, "ASIC_STATE:SAI_OBJECT_TYPE_MY_SID_ENTRY")
+        (status, fvs) = tbl.get(my_sid)
+        assert status == True
+        for fv in fvs:
+            if fv[0] == "SAI_MY_SID_ENTRY_ATTR_ENDPOINT_BEHAVIOR":
+                assert fv[1] == "SAI_MY_SID_ENTRY_ENDPOINT_BEHAVIOR_DT6"
+            if fv[0] == "SAI_MY_SID_ENTRY_ATTR_VRF":
+                assert fv[1] == self.vrf_id
+
+        # remove srv6 mysid end.dt6 behavior
+        dvs.runcmd("ip -6 route del fc00:0:0:1:6b::/128 encap seg6local action End.DT6 vrftable {} dev sr0".format(self.vrf_table_id))
+
+        # check application database
+        self.pdb.wait_for_deleted_entry("SRV6_MY_SID_TABLE", "40:24:16:0:fc00:0:0:1:6b::")
+
+        # verify that the mysid has been removed from the ASIC
+        self.adb.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_MY_SID_ENTRY", len(self.initial_my_sid_entries))
+
+        # unconfigure srv6 locator
+        dvs.runcmd("vtysh -c \"configure terminal\" -c \"segment-routing\" -c \"no srv6\"")
+
+        self.teardown_srv6(dvs)
+
+    def test_AddRemoveSrv6MySidUDT6(self, dvs, testlog):
+
+        _, output = dvs.runcmd(f"vtysh -c 'show zebra dplane providers'")
+        if 'dplane_fpm_sonic' not in output:
+            pytest.skip("'dplane_fpm_sonic' required for this test is not available, skipping", allow_module_level=True)
+
+        self.setup_srv6(dvs)
+
+        # configure srv6 usid locator
+        dvs.runcmd("vtysh -c \"configure terminal\" -c \"segment-routing\" -c \"srv6\" -c \"locators\" -c \"locator loc1\" -c \"prefix fc00:0:2::/48 block-len 32 node-len 16 func-bits 16\" -c \"behavior usid\"")
+
+        # create srv6 mysid udt6 behavior
+        dvs.runcmd("ip -6 route add fc00:0:2:ff05::/128 encap seg6local action End.DT6 vrftable {} dev sr0".format(self.vrf_table_id))
+
+        # check application database
+        self.pdb.wait_for_entry("SRV6_MY_SID_TABLE", "32:16:16:0:fc00:0:2:ff05::")
```

Comment:
Could you check Endpoint behavior of MYSID entry in APPL_DB?

Reply:
Done. I added some checks to verify endpoint behavior + attributes of the MYSID entries in APPL_DB